extensions/markdown-language-features/src/extension.ts

```diff
@@ -45,6 +47,43 @@ export function activate(context: vscode.ExtensionContext) {
 		logger.updateConfiguration();
 		previewManager.updateConfiguration();
 	}));
+
+	context.subscriptions.push(vscode.workspace.onWillDropOnTextEditor(e => {
+		e.waitUntil((async () => {
```

Comment:
Since `DataTransfer` has async methods, I think almost every caller will have to use `waitUntil` which is quite ugly. Will see if we can figure out a better approach to this